### PR TITLE
decomp scrcmd_6 based off of tgsm's work

### PR DESCRIPTION
**Pull request by SethBarberee**
Based off of @tgsm's prior work set in discord, finished up the module.

### CODE REVIEW

**Review by ProjectRevoTPP — DISMISSED** [review]

File: arm9/src/scrcmd_sound.c [hunk 1]
```diff
@@ -0,0 +1,299 @@
+#include "global.h"
+#include "script.h"
```
Comment (tgsm) [inline arm9/src/scrcmd_sound.c:2]: scrcmd.h already includes both of these, so they don't have to be here

File: arm9/src/scrcmd_sound.c [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+extern BOOL FUN_02041598(struct ScriptContext* ctx);
+extern BOOL FUN_020415BC(struct ScriptContext* ctx);
+extern BOOL FUN_020415CC(struct ScriptContext* ctx);
+extern BOOL ScrCmd_playbgm(struct ScriptContext* ctx);
+extern BOOL FUN_020415E0(struct ScriptContext* ctx);
+extern BOOL FUN_02041618(struct ScriptContext* ctx);
+extern BOOL FUN_02041644(struct ScriptContext* ctx);
+extern BOOL FUN_02041650(struct ScriptContext* ctx);
+extern BOOL FUN_02041664(struct ScriptContext* ctx);
+extern BOOL FUN_02041674(struct ScriptContext* ctx);
```
Comment (tgsm) [inline arm9/src/scrcmd_sound.c:31]: These should go into scrcmd.h, starting at line 153 (also make sure to add a comment saying these are from scrcmd_sound)

File: arm9/src/scrcmd_sound.c [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+    FUN_0200538C(0x7F, unk, 0);
+    SetupNativeScript(ctx, FUN_02041464);
+
+    return TRUE;
+}
+
+THUMB_FUNC BOOL FUN_0204149C(struct ScriptContext* ctx)
+{
+
+    u8 PtrIndx0 = *ctx->scriptPtr++;
```
Comment (tgsm) [inline arm9/src/scrcmd_sound.c:134]: We have a macro for these :)
```suggestion
    u8 PtrIndx0 = ScriptReadByte(ctx);
```

File: arm9/src/scrcmd_sound.c [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+    u8 PtrIndx1 = *ctx->scriptPtr++;
```
Comment (tgsm) [inline arm9/src/scrcmd_sound.c:136]: ditto

File: arm9/src/scrcmd_sound.c [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+THUMB_FUNC BOOL FUN_020414B8(struct ScriptContext *ctx)
+{
+    FUN_02005308(4, ScriptReadHalfword(ctx));
+    return FALSE;
+}
+
+THUMB_FUNC BOOL FUN_020414CC(struct ScriptContext* ctx)
+{
+    u8 PtrIndx0 = *ctx->scriptPtr++;
```
Comment (tgsm) [inline arm9/src/scrcmd_sound.c:150]: ditto

File: arm9/src/scrcmd_sound.c [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+    ctx->data[0] = unk;
+    SetupNativeScript(ctx, FUN_02041540);
+
+    return TRUE;
+}
+
+THUMB_FUNC BOOL FUN_02041540(struct ScriptContext* ctx)
+{
+    if(!FUN_02005508((u16)ctx->data[0]))
+        return 1;
```
Comment (tgsm) [inline arm9/src/scrcmd_sound.c:183]:
```suggestion
        return TRUE;
```

File: arm9/src/scrcmd_sound.c [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+    else
+        return 0;
```
Comment (tgsm) [inline arm9/src/scrcmd_sound.c:185]:
```suggestion
        return FALSE;
```

**Review by tgsm — CHANGES_REQUESTED** [review]
some minor nits

**Review by ProjectRevoTPP — APPROVED** [review]

**Review by tgsm — DISMISSED** [review]

File: arm9/src/scrcmd_sound.c [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+extern void FUN_020040F4(u8);
+extern BOOL FUN_02005508(u16);
+extern void FUN_02005578(u16);
+extern BOOL FUN_02005670(void);
+extern void FUN_0200488C(u16, u16);
+
+
+THUMB_FUNC BOOL FUN_020413C0(struct ScriptContext *ctx)
+{
+    u16 unk0;
```
Comment (red031000) [inline arm9/src/scrcmd_sound.c:33]: these declarations and assignments can be combined into one

File: arm9/src/scrcmd_sound.c [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+extern void FUN_020054F0(u16, u32);
+extern void FUN_020047C8(u8, u8);
+extern void FUN_020040F4(u8);
+extern BOOL FUN_02005508(u16);
+extern void FUN_02005578(u16);
+extern BOOL FUN_02005670(void);
+extern void FUN_0200488C(u16, u16);
+
+
+THUMB_FUNC BOOL FUN_020413C0(struct ScriptContext *ctx)
```
Comment (red031000) [inline arm9/src/scrcmd_sound.c:31]: these funcs should all be named on their scrcmd index instead of keeping FUN_

File: arm9/src/scrcmd_sound.c [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+
+THUMB_FUNC BOOL FUN_02041558(struct ScriptContext* ctx)
+{
+    u16 unk0 = VarGet(ctx->unk80, ScriptReadHalfword(ctx));
+    u16 unused = VarGet(ctx->unk80, ScriptReadHalfword(ctx));
+    FUN_02005578(unk0);
+    return FALSE;
+}
+
+THUMB_FUNC BOOL ScrCmd_waitcry(struct ScriptContext* ctx)
```
Comment (red031000) [inline arm9/src/scrcmd_sound.c:166]: ScrCmd_PlayCryWait would be a better name (note CamelCase for commands)

File: arm9/src/scrcmd_sound.c [hunk 11]
```diff
@@ -0,0 +1,269 @@
...
+{
+#pragma unused(ctx)
+
+    if(!FUN_02005670())
+        return TRUE;
+    else
+        return FALSE;
+}
+
+THUMB_FUNC BOOL ScrCmd_playbgm(struct ScriptContext* ctx)
```
Comment (red031000) [inline arm9/src/scrcmd_sound.c:182]: ScrCmd_PlayBgm

**Review by red031000 — CHANGES_REQUESTED** [review]

**Review by red031000 — APPROVED** [review]

**Pull request merged**